accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,390 @@
+
+For now the Maintenance Mode is a method called on a system object which lookup the schedule and check
+if we are currently inside of a defined maintenance window or not. There maybe some functionalities
+which require this information.
+
+Putting a system manual into maintenance mode is a feature of the second iteration.
+
+
+### Maintenance Window
+
+A Maintenance Window specify a time when changes on a system can be applied.
+- It has a start date-time and an end date-time.
+- It can be defined as recurring interval (RRULE).
+- Exceptions should be supported (EXDATE)
+- Events for a full day (anniversary) or multiple days must be considered.
+- The Summary may define the schedule name (See also Maintenance Schedule).
+
+For practical reasons the window describes when an action can be `started`.
+As it is unpredictable to know how long an action will take, we cannot calculate,
+if the action can also be `finished` inside of the Maintenance Window.
+
+A started Action should always be finished to not risk a broken system.
+
+#### Action Chains
+
+Action Chains with reboots are split into multiple actions.
+As the goal is, that everything what was started inside of a maintenance window should also be finished,
+we need to finish whole action chains also after the window closed.
+
+If we won't do it, we would end up with broken systems.
+
+
+### Maintenance Schedule
+
+A Maintenance Schedule is a set of Maintenance Windows. A Maintenance Schedule has a name/label.
+The data should be stored as ICalendar data (https://tools.ietf.org/html/rfc5545).
+In the first iteration importing the data created by an external tool is required.
+Uyuni will not provide a User interface for generating or manipulating Maintenance Windows.
+
+This is a task for future iterations.
+
+To parse the windows we need a library for ICalendar files. E.g. ical4j(https://github.com/ical4j/ical4j).
+
+There could be two types of Schedules for the first iteration
+
+1. One ICalendar file represent one schedule
+
+Every `VEVENT` describe a maintenance window.
+
+2. One ICalendar file contains multiple schedules
+
+Every `VEVENT` must have a `SUMMARY`. All events with the same "summary" belong to the same schedule.
+The summary must match the schedule name in the Uyuni Database.
+
+### Adding a new Maintenance Schedule
+
+In the left sidebar a new page should be added below `Schedule`.
+In this page you can see a list of existing schedules. You can edit them, when you select an existing schedule.
+There should be a "Create Maintenance Schedule" link in the upper left corner which open a new page where the following values can be specified:
+
+- Schedule Name
+- Schedule Type (Single, Multi, ....)
+- Data (ICalendar content) with file upload option
+
+A "Create" button will create the new schedule. The ICalendar data are stored in the database.
+
+Adding a Maintenance Schedule via XMLRPC API must be possible as well.
+
+
+### Updating a Maintenance Schedule
```

Comment:
```suggestion
### Updating/Deleting a Maintenance Schedule
```

Reply:
Removing a whole schedule is easy, as it does not invalidate any scheduled action. No schedule means permanent maintenance mode. You can execute actions all the time.

But I will add this info to this section.